src/Features/CSharp/Portable/ReplaceDefaultLiteral/CSharpReplaceDefaultLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,150 @@
+                var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+                var (newExpression, displayText) = GetReplacementExpressionAndText(
+                    context.Document, defaultLiteral, semanticModel, context.CancellationToken);
+
+                if (newExpression != null)
+                {
+                    context.RegisterCodeFix(
+                        new MyCodeAction(
+                            c => ReplaceAsync(context.Document, context.Span, newExpression, c),
+                            displayText),
+                        context.Diagnostics);
+                }
+            }
+        }
+
+        private static async Task<Document> ReplaceAsync(
+            Document document, TextSpan span, SyntaxNode newExpression, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var defaultToken = syntaxRoot.FindToken(span.Start);
+            var defaultLiteral = (LiteralExpressionSyntax)defaultToken.Parent;
+
+            var newRoot = syntaxRoot.ReplaceNode(defaultLiteral, newExpression.WithTriviaFrom(defaultLiteral));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private static (SyntaxNode newExpression, string displayText) GetReplacementExpressionAndText(
+            Document document,
+            LiteralExpressionSyntax defaultLiteral,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var type = semanticModel.GetTypeInfo(defaultLiteral, cancellationToken).ConvertedType;
+            if (type != null && type.TypeKind != TypeKind.Error)
+            {
+                if (IsFlagsEnum(type, semanticModel.Compilation) &&
+                    type.GetMembers("None").FirstOrDefault() is IFieldSymbol field && IsZero(field.ConstantValue))
+                {
+                    return GenerateMemberAccess("None");
+                }
+                else if (type.Equals(semanticModel.Compilation.GetTypeByMetadataName("System.Threading.CancellationToken")))
+                {
+                    return GenerateMemberAccess(nameof(CancellationToken.None));
+                }
+                else if (type.SpecialType == SpecialType.System_IntPtr || type.SpecialType == SpecialType.System_UIntPtr)
+                {
+                    return GenerateMemberAccess(nameof(IntPtr.Zero));
+                }
+                else if (semanticModel.GetConstantValue(defaultLiteral, cancellationToken) is var constant && constant.HasValue)
+                {
+                    var newLiteral = generator.LiteralExpression(constant.Value);
+                    return (newLiteral, newLiteral.ToString());
+                }
+                else if (!type.ContainsAnonymousType())
```

Comment:
i didn't see the ! for a second, and i was so confused as to what was going on :)